front/src/common/Map/Search/MapSearchOperationalPoint.tsx

```diff
@@ -0,0 +1,188 @@
+  }, [searchResults, chCodeFilter, mainOperationalPointsOnly]);
+
+  const onResultClick = (result: SearchResultItemOperationalPoint) => {
+    onResultSearchClick({
+      result,
+      map,
+      updateExtViewport,
+      dispatch,
+      title: result.name,
+      setSearchTerm,
+    });
+    closeMapSearchPopUp();
+  };
+
+  return (
+    <>
+      <div className="d-flex mb-2 flex-column flex-md-row">
+        <span className="flex-fill col-md-6 col-xl-7 pl-0 mb-2">
+          <InputSNCF
+            id="map-search-operational-points"
+            name="map-search-operational-points"
+            placeholder={t('placeholdername')}
+            title={t('placeholdername')}
+            type="text"
+            value={searchTerm}
+            onChange={(e) => {
+              setSearchTerm(e.target.value);
+            }}
+            onClear={() => setSearchTerm('')}
+            clearButton
+            noMargin
+            sm
+            focus
+          />
+        </span>
+        <span className="col-md-3 pl-0 mb-2">
+          <InputSNCF
+            id="map-search-operational-points-ch-code"
+            type="text"
+            placeholder={t('placeholderchcode')}
+            onChange={(e) => {
+              setChCodeFilter(e.target.value);
+            }}
+            onClear={() => setChCodeFilter('')}
+            value={chCodeFilter}
+            disabled={mainOperationalPointsOnly}
+            clearButton
+            noMargin
+            sm
+          />
+        </span>
+        <span className="col-md-3 col-xl-2 pr-2 pl-0 mt-md-1">
+          <CheckboxRadioSNCF
+            id="map-search-operational-points-main-only"
+            type="checkbox"
+            label={t('mainOperationalPointsOnly')}
+            checked={mainOperationalPointsOnly}
+            onChange={() => setMainOperationalPointsOnly(!mainOperationalPointsOnly)}
+          />
+        </span>
+      </div>
+      <h2 className="text-center mt-3">
+        {searchResults.length > 100
+          ? t('resultsCountTooMuch')
+          : t('resultsCount', {
+              count: orderedResults.length,
+            })}
+      </h2>
+      <div className="search-results">
+        {searchResults.length > 0 &&
+          searchResults.length <= 100 &&
+          orderedResults.map((searchResult) => (
+            <button
+              type="button"
+              className={cx('operational-point', { main: searchResult.ch === '' })}
+              key={`mapSearchOperationalPoint-${searchResult.obj_id}`}
+              onClick={() => onResultClick(searchResult)}
+            >
+              <span className="trigram">{searchResult.trigram}</span>
+              <span className="name">
+                {searchResult.name}
+                <span className="ch">{searchResult.ch}</span>
+              </span>
+              <span className="uic">{searchResult.uic}</span>
+            </button>
+          ))}
+      </div>
```

Comment:
To split component responsibility (and reduce component size), I think we should still keep another component who could be called  `<MapSearchResults />` that would be only responsible to display the results list. When mapsearchline and mapsearchsignal will be fixed, we might be able to reuse it for them as well.

Good practice would be also to have a `<OperationalPointCard />` component with just the <button> logic inside it.

The OnResultClick() function could be moved inside it (or inside MapSearchResults if we don't use OperationalPointCard)

Reply:
We discussed about it, I think considering
- small size
- logic-less of the buttons
- no able to reuse (buttons change according to search object)
we can let thoses inside component. Thx anyway!